lib/appium/helpers/util.js

```diff
@@ -0,0 +1,174 @@
+        var findSimCommand = 'cordova run --list --emulator | grep ^iPhone | tail -n1';
+
+        medicLog('running:');
+        medicLog('    ' + findSimCommand);
+
+        var findSimResult = shelljs.exec(findSimCommand, { silent: true, async: false });
+
+        if (findSimResult.code > 0) {
+            medicLog('Failed to find simulator we deployed to');
+            return;
+        }
+
+        var split = findSimResult.output.split(', ');
+        // Format of the output is "iPhone-6s-Plus, 9.1"
+        // Extract the device name and the version number
+        var device = split[0].replace(/-/g, ' ').trim();
+        var version = split[1].trim();
+
+        // Next, figure out the ID of the simulator we found
+        var instrCommand = 'instruments -s devices | grep ^iPhone';
+        medicLog('running:');
+        medicLog('    ' + instrCommand);
+
+        var instrResult = shelljs.exec(instrCommand, { silent: true, async: false });
+
+        if (instrResult.code > 0) {
+            medicLog('Failed to get the list of simulators');
+            return;
+        }
+
+        // This matches <device> (<version>) [<simulator-id>]
+        // ... or <device> (<version>) [<simulator-id>] (Simulator) in Xcode 7.3
+        var simIdRegex = /^([a-zA-Z\d ]+) \(([\d.]+)\) \[([a-zA-Z\d\-]*)\].*$/;
+
+
+        var simId = null;
+        var lines = instrResult.output.split(/\n/);
+        lines.forEach(function(line) {
+            var simIdMatch = simIdRegex.exec(line);
+            if (simIdMatch && simIdMatch.length === 4 && simIdMatch[1] === device && simIdMatch[2] === version) {
+                simId = encodeURIComponent(simIdMatch[3]);
+            }
+        });
+
+        return simId;
+    }
+
+    function doesFileExist(filePath) {
+        var fileExists = false;
+        try {
+            fs.statSync(filePath);
+            fileExists = true;
+        } catch (e) {
+            fileExists = false;
+        }
+        return fileExists;
+    }
+
+    function getSqlite3InsertionCommand(destinationTCCFile, service, appName) {
+        return util.format('sqlite3 %s "insert into access' +
+            '(service, client, client_type, allowed, prompt_count, csreq) values(\'%s\', \'%s\', ' +
+            '0,1,1,NULL)"', destinationTCCFile, service, appName);
+    }
+
+    return {
+
+        // constants
+        ANDROID: 'android',
+        BLACKBERRY: 'blackberry10',
+        IOS: 'ios',
+        WINDOWS: 'windows',
+
+        DEFAULT_ENCODING: 'utf-8',
+        DEFAULT_LOG_TIME: 15,
+        DEFAULT_LOG_TIME_ADDITIONAL: 2,
+
+        // functions
+        fatal: fatal,
+        isWindows: isWindows,
+        medicLog: medicLog,
+        contains: contains,
+        secToMin: secToMin,
+        countAndroidDevices: countAndroidDevices,
+        getSimulatorsFolder: getSimulatorsFolder,
+        getSimId: getSimId,
+        doesFileExist: doesFileExist,
+        getSqlite3InsertionCommand: getSqlite3InsertionCommand
```

Comment:
Looks like some of these methods are not used anywhere. Could you please remove unused ones?